src/rgw/rgw_rados.cc

```diff
@@ -4183,7 +4247,7 @@ void RGWRados::build_bucket_index_marker(const string& shard_id_str, const strin
 
 int RGWRados::open_bucket_index_ctx(rgw_bucket& bucket, librados::IoCtx& index_ctx)
 {
-  int r = open_pool_ctx(bucket.index_pool, index_ctx);
+  int r = open_pool_ctx(bucket.index_pool, index_ctx, bucket.ns);
```

Comment:
i agree - you can't use `bucket.ns` with `bucket.index_pool` as currently designed. i'm actually worried about using it with `data_pool` and `data_extra_pool` too, now that I'm reading through `RGWRados::set_bucket_location_by_rule()`

those pool names are coming from the zone's `RGWZonePlacementInfo`, which i think would also need to be parsed as `pool:namespace`. then it looks like `rgw_bucket` needs to store a separate namespace for each of those pools, right? hopefully @yehudasa's bucket/pool cleanup can help with this